### PR TITLE
Add delete functionality for old repositories

### PR DESCRIPTION
**Pull request by SvetoslavDe**

### CODE REVIEW

File: code-execution/src/test/java/com/dojo/codeexecution/service/GitManagerTest.java [hunk 1]
```diff
@@ -149,4 +156,62 @@ public void buildParentWebHookWhenIsHookAvailable() throws IOException {
         verify(gitConfig, times(1)).getUser();
         verify(gitConfig, times(1)).getParentRepositoryName();
     }
+
+    @Test
+    public void testDeleteAllReposForSpecificGame() throws IOException{
+        when(ghRepository.getName()).thenReturn("gamified-hiring-user1-kata");
```
Comment (SlavinaKlisarska) [inline code-execution/src/test/java/com/dojo/codeexecution/service/GitManagerTest.java:162]: can we extract data setup in a separate method

File: code-execution/src/test/java/com/dojo/codeexecution/service/GitManagerTest.java [hunk 2]
```diff
@@ -149,4 +156,62 @@ public void buildParentWebHookWhenIsHookAvailable() throws IOException {
...
+        GHRepository repo2 = mock(GHRepository.class);
+        when(repo2.getName()).thenReturn("gamified-hiring-user2-kata");
```
Comment (SlavinaKlisarska) [inline code-execution/src/test/java/com/dojo/codeexecution/service/GitManagerTest.java:164]: extract constants when using string literals

File: code-execution/src/test/java/com/dojo/codeexecution/service/GitManagerTest.java [hunk 3]
```diff
@@ -149,4 +156,62 @@ public void buildParentWebHookWhenIsHookAvailable() throws IOException {
...
+        when(myself.getAllRepositories()).thenReturn(repoMap);
+        String expectedResult = "Following repositories have been deleted: " + System.lineSeparator() +
+                "gamified-hiring-user2-kata" + System.lineSeparator()+
+                "gamified-hiring-user1-kata";
+        assertEquals(expectedResult, gitManager.deleteReposForParticularGame("kata"));
+    }
+
+    @Test
+    public void TestRepositoryDeletion_OnlyReposWithQualifyingNameShouldBeDeleted() throws Exception{
+        Map<String, GHRepository> allRepos = new HashMap<>();
```
Comment (SlavinaKlisarska) [inline code-execution/src/test/java/com/dojo/codeexecution/service/GitManagerTest.java:181]: same for data setup - method is too big

File: code-execution/src/main/java/com/dojo/codeexecution/service/GitManager.java [hunk 4]
```diff
@@ -81,4 +86,27 @@ private String getRepositoryNameByOwner(String username, String game) throws IOE
     private GHUser getGitHubUser(String username) throws IOException {
         return gitHub.getUser(username);
     }
+
+    public String deleteReposForParticularGame(String game) throws IOException {
+        GHMyself account = gitHub.getMyself();
+        Collection<GHRepository> listOfAllRepos = account.getAllRepositories().values();
+        List<String> nameOfDeletedRepos = new CopyOnWriteArrayList<>();
```
Comment (SlavinaKlisarska) [inline code-execution/src/main/java/com/dojo/codeexecution/service/GitManager.java:93]: deletedRepos would be enough for naming the list or deletedReposNames is also acceptable

File: code-execution/src/main/java/com/dojo/codeexecution/service/GitManager.java [hunk 5]
```diff
@@ -81,4 +86,27 @@ private String getRepositoryNameByOwner(String username, String game) throws IOE
     private GHUser getGitHubUser(String username) throws IOException {
         return gitHub.getUser(username);
     }
+
+    public String deleteReposForParticularGame(String game) throws IOException {
+        GHMyself account = gitHub.getMyself();
+        Collection<GHRepository> listOfAllRepos = account.getAllRepositories().values();
```
Comment (SlavinaKlisarska) [inline code-execution/src/main/java/com/dojo/codeexecution/service/GitManager.java:92]: listOfAllRepos can be just reposm, imagine if we did everywhere setof..., mapof.., etc.

File: code-execution/src/main/java/com/dojo/codeexecution/service/GitManager.java [hunk 6]
```diff
@@ -81,4 +86,27 @@ private String getRepositoryNameByOwner(String username, String game) throws IOE
...
+                .filter(repo->repo.getName().startsWith(REPO_PREFIX) && repo.getName().endsWith("-"+game))
+                .forEach(repo-> {
+                    try {
+                        repo.delete();
+                        nameOfDeletedRepos.add(repo.getName());
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                });
+        if(nameOfDeletedRepos.size() == 0) {
```
Comment (SlavinaKlisarska) [inline code-execution/src/main/java/com/dojo/codeexecution/service/GitManager.java:104]: this is already a second thing this method is doing, processing the result

**Review by SlavinaKlisarska — CHANGES_REQUESTED** [review]

File: code-execution/src/test/java/com/dojo/codeexecution/service/GitManagerTest.java [hunk 7]
```diff
@@ -149,4 +161,71 @@ public void buildParentWebHookWhenIsHookAvailable() throws IOException {
...
+                when(repo.getName()).thenReturn(RANDOM_REPO_NAME + i);
+                allRepos.put(repo.getName(), repo);
+                reposToKeep.put(repo.getName(), repo);
+            }
+
+            when(gitHub.getMyself()).thenReturn(myself);
+            when(myself.getAllRepositories()).thenReturn(allRepos);
+        }
+
+        @org.junit.jupiter.api.Test
```
Comment (SlavinaKlisarska) [inline code-execution/src/test/java/com/dojo/codeexecution/service/GitManagerTest.java:202]: why is this imported differntly

File: code-execution/src/main/java/com/dojo/codeexecution/service/GitManager.java [hunk 8]
```diff
@@ -81,4 +85,32 @@ private String getRepositoryNameByOwner(String username, String game) throws IOE
...
+                        repo.delete();
+                        deletedRepos.add(repo.getName());
+                    } catch (IOException e) {
+                        e.printStackTrace();
+                    }
+                });
+        return generateResult(deletedRepos);
+    }
+
+    private String generateResult(List<String> deletedRepos) {
```
Comment (SlavinaKlisarska) [inline code-execution/src/main/java/com/dojo/codeexecution/service/GitManager.java:106]: GitManager is now responsible for more than one thing which is breaking the single responsibility principle;
generating a result should be done by a separate class

**Review by SlavinaKlisarska — CHANGES_REQUESTED** [review]

File: code-execution/src/test/java/com/dojo/codeexecution/service/GitManagerTest.java (hunk 7 above)
Comment (SvetoslavDe) [reply code-execution/src/test/java/com/dojo/codeexecution/service/GitManagerTest.java:202]: @Test from junit4 doesn't work with @nested test class, so @Test from JunitJupiter had to be used

**Review by SlavinaKlisarska — APPROVED** [review]

**Pull request closed without merging**